dm/syncer/syncer.go

```diff
@@ -1598,6 +1603,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 			currentLocation = shardingReSync.currLocation
 			// if suffix>0, we are replacing error
 			s.isReplacingErr = currentLocation.Suffix != 0
+			s.locations.reset(shardingReSync.currLocation)
```

Comment:
do we need to reset `s.locations` when `closeShardingResync` and `isReplacingErr` ?

Reply:
can't understand

Reply:
If I understand correctly, this PR is intended to maintain information about the location of the current binlog stream in `s.locations` and to compare it each time when `GetEvent`

So do you also need to update the location in `s.locations` when `closeShardingResync` or `ReplacingErr`?

Reply:
line1555 is inside closeShardingResync